devito/types/dense.py

```diff
@@ -1311,3 +1312,198 @@ def parent(self):
         return self._parent
 
     _pickle_kwargs = Function._pickle_kwargs + ['parent']
+
+
+class SeparableFunction(Differentiable):
+    """
+    A function separable in its dimensions, ie f(x,y,z) = f(x)*f(y)*f(z)
+
+    Takes the same parameters as Function and
+
+    Parameters
+    ----------
+        sep: 'sum' or 'prod'
```

Comment:
de-indent

the type is str

values (sum, prod) should be discussed in the description